### PR TITLE
update sidecars to latest in CSI deployment

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Additional Note for reviewer:
- [x] all sidecar controllers are updated to the latest version
 
* provisioner: 3.3.0
 * resizer: 1.6.0
 * snapshotter: 6.1.0
 * attacher: 4.0.0
  
- [x] the default fstype has been set to "ext4" as its a requirement with the latest attacher sidecar update 

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by humblec** [comment]
Cc @travisn

File: Documentation/Helm-Charts/operator-chart.md [hunk 1]
```diff
@@ -120,9 +120,9 @@ The following tables lists the configurable parameters of the rook-operator char
 | `csi.nfsPluginUpdateStrategy`       | CSI NFS plugin daemonset update strategy, supported values are OnDelete and RollingUpdate.                                                                | `RollingUpdate`                                                |
 | `csi.registrar.image`               | Kubernetes CSI registrar image.                                                                                                                           | `registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.5.1` |
 | `csi.resizer.image`                 | Kubernetes CSI resizer image.                                                                                                                             | `registry.k8s.io/sig-storage/csi-resizer:v1.4.0`               |
```
Comment (travisn) [inline Documentation/Helm-Charts/operator-chart.md:122]:
```suggestion
| `csi.resizer.image`                 | Kubernetes CSI resizer image.                                                                                                                             | `registry.k8s.io/sig-storage/csi-resizer:v1.6.0`               |
```

File: pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml [hunk 2]
```diff
@@ -33,6 +33,7 @@ spec:
             - "--leader-election-lease-duration=137s"
             - "--leader-election-renew-deadline=107s"
             - "--leader-election-retry-period=26s"
+            - "--default-fstype=ext4"
```
Comment (travisn) [inline pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml:36]: This would be overridden by the fstype in the storage class?

**Review by travisn — CHANGES_REQUESTED** [review]
Looks like one more spot to update

File: pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml (hunk 2 above)
Comment (humblec) [reply pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml:36]: previously the attacher sidecar always started with `ext4` as the default value and from v4.0.0 onwards the default is `""/nil.` https://github.com/kubernetes-csi/external-attacher/pull/358/files#diff-f422861635f15f707a5f60c81e05a96a541169e64b5f2cae6e0e964f7181b640R69 . Hence the CSI drivers that depended on `ext4` as the default need to change their deployments to explicitly set `--default-fstype=ext4`. The SC fstype setting will be respected by the provisioner and the set value will be send by the provisioner as part of the provision request.

**Review by travisn — APPROVED** [review]

**Pull request merged**